File: src/llm/plugins/openrouter.py
import logging
import typing as T
from enum import Enum

from pydantic import BaseModel, Field

from llm import LLMConfig
from llm.openai_compatible import OpenAICompatibleLLM

R = T.TypeVar("R", bound=BaseModel)


class OpenRouterModel(str, Enum):
    """Available OpenRouter models."""

    ANTHROPIC_SONNET_4_5 = "anthropic/claude-sonnet-4.5"
    ANTHROPIC_OPUS_4_5 = "anthropic/claude-opus-4.5"
    ANTHROPIC_HAIKU_4_5 = "anthropic/claude-haiku-4.5"
    MOONSHOT_KIMI_K2_5 = "moonshotai/kimi-k2.5"
    MINIMAX_M2_1 = "minimax/minimax-m2.1"
    Z_AI_GLM_4_7 = "z-ai/glm-4.7"
    X_AI_GROK_4_FAST = "x-ai/grok-4-fast"
    DEEPSEEK_V3_2 = "deepseek/deepseek-v3.2"
    LLAMA_3_3_70B = "meta-llama/llama-3.3-70b-instruct"


class OpenRouterConfig(LLMConfig):
    """OpenRouter-specific configuration with model enum."""

    base_url: T.Optional[str] = Field(
        default="https://api.openmind.org/api/core/openrouter",
        description="Base URL for the OpenRouter API endpoint",
    )
    model: T.Optional[T.Union[OpenRouterModel, str]] = Field(
        default=OpenRouterModel.ANTHROPIC_SONNET_4_5,
        description="OpenRouter model to use",
    )


class OpenRouter(OpenAICompatibleLLM[R]):
    """OpenRouter LLM implementation using OpenAI-compatible API."""

    DEFAULT_MODEL = "meta-llama/llama-3.3-70b-instruct"
    PROVIDER_NAME = "OpenRouter"
    LOG_LEVEL = logging.INFO

File: plugins/llm/openrouter.go
package llm

import (
	"github.com/openmind/om1/internal/llm"
)

func init() {
	llm.Register("OpenRouter", NewOpenRouter)
}

type openRouterModel = string

const (
	OpenRouterModelAnthropicSonnet45 openRouterModel = "anthropic/claude-sonnet-4.5"
	OpenRouterModelAnthropicOpus45   openRouterModel = "anthropic/claude-opus-4.5"
	OpenRouterModelAnthropicHaiku45  openRouterModel = "anthropic/claude-haiku-4.5"
	OpenRouterModelMoonshotKimiK25   openRouterModel = "moonshotai/kimi-k2.5"
	OpenRouterModelMinimaxM21        openRouterModel = "minimax/minimax-m2.1"
	OpenRouterModelZAIGLM47          openRouterModel = "z-ai/glm-4.7"
	OpenRouterModelXAIGrok4Fast      openRouterModel = "x-ai/grok-4-fast"
	OpenRouterModelDeepSeekV32       openRouterModel = "deepseek/deepseek-v3.2"
	OpenRouterModelLlama3370B        openRouterModel = "meta-llama/llama-3.3-70b-instruct"

	defaultOpenRouterModel   openRouterModel = OpenRouterModelAnthropicSonnet45
	defaultOpenRouterBaseURL string          = "https://api.openmind.com/api/core/openrouter"
)

// NewOpenRouter creates an OpenRouter LLM using the OpenAI-compatible API.
func NewOpenRouter(configMap map[string]any) (llm.LLM, error) {
	return newOpenAICompat("OpenRouter", configMap, defaultOpenRouterModel, defaultOpenRouterBaseURL, "auto", true)
}
